src/browserlib/extract-cddl.mjs

```diff
@@ -0,0 +1,125 @@
+import getCodeElements from './get-code-elements.mjs';
+import trimSpaces from './trim-spaces.mjs';
+
+/**
+ * Extract the list of CDDL definitions in the current spec.
+ *
+ * A spec may define more that one CDDL module. For example, the WebDriver BiDi
+ * spec has CDDL definitions that apply to either of both the local end and the
+ * remote end. The functions returns an array that lists all CDDL modules.
+ * 
+ * Each CDDL module is represented as an object with the following keys whose
+ * values are strings:
+ * - shortname: the CDDL module shortname. Shortname is "" if the spec does not
+ * define any module, and "all" for the dump of all CDDL definitions.
+ * - label: A full name for the CDDL module, when defined.
+ * - cddl: A dump of the CDDL definitions.
+ *
+ * If the spec defines more than one module, the first item in the array is the
+ * "all" module that contains a dump of all CDDL definitions, regardless of the
+ * module they are actually defined for (the assumption is that looking at the
+ * union of all CDDL modules defined in a spec will always make sense, and that
+ * a spec will never reuse the same rule name with a different definition for
+ * different CDDL modules).
+ *
+ * @function
+ * @public 
+ * @return {Promise} The promise to get a dump of the CDDL definitions per
```

Comment:
doesn't seem to be returning a Promise?

Reply:
Indeed. And that's also a copy-and-paste from IDL extraction (which also incorrectly says it returns a Promise). I guess that shows these comments are not particularly useful since we don't use them. Actual typing would be more useful. But that's a topic for another day. I'll fix the comments in the meantime.